### PR TITLE
feat: improve display supported blockchains [LIVE-8533]

### PR DESCRIPTION
**Pull request by Wozacosta**
<details><summary>try the live app with this manifest </summary>

```json
{
  "id": "ledger-wallet-connect",
  "name": "WalletConnect local",
  "url": "https://wallet-connect-live-cqfp0okhv-ledgerhq.vercel.app/",
  "params": {},
  "homepageUrl": "https://walletconnect.com/",
  "icon": "https://forum.zeroqode.com/uploads/default/original/2X/e/e363c6521db27335d44c1134d230b8992792dde4.png",
  "platform": "all",
  "apiVersion": "^2.0.0",
  "manifestVersion": "1",
  "branch": "stable",
  "categories": ["bridge", "defi"],
  "currencies": [
    "ethereum",
    "polygon",
    "bsc",
    "optimism",
    "arbitrum",
    "avalanche_c_chain",
    "base",
    "elrond",
    "ethereum_sepolia",
    "arbitrum_sepolia",
    "optimism_sepolia",
    "base_sepolia",
    "ethereum_holesky"
  ],
  "content": {
    "shortDescription": {
      "en": "WalletConnect is an open source protocol for connecting decentralised applications to mobile wallets with QR code scanning or deep linking. V2 introduces new features, including the ability to connect to multiple dapps in parallel with multiple accounts. It's important to note that not all dapps currently support V2"
    },
    "description": {
      "en": "WalletConnect is an open source protocol for connecting decentralised applications to mobile wallets with QR code scanning or deep linking. V2 introduces new features, including the ability to connect to multiple dapps in parallel with multiple accounts. It's important to note that not all dapps currently support V2"
    }
  },
  "permissions": [
    "account.list",
    "account.request",
    "message.sign",
    "transaction.sign",
    "transaction.signAndBroadcast",
    "wallet.userId",
    "wallet.info"
  ],
  "domains": ["http://*", "https://*"]
}

```
</details>

### 📝 Description

_Replace this text by a clear and concise description of what this pull request is about and why it is needed._ 

The PR is following the issue specs with some notable differences listed here: 

- Confirm / Reject buttons are sticky but with a blurry transparent background. 

https://github.com/LedgerHQ/wallet-connect-live-app/assets/5050709/6ddf578e-721a-459a-8ee6-4ee4f4b48037 

- On the proposal screen, Accounts are separated by a row representing the chain, that chain row gives a visual feedback and indication that it is a required chain or not 

https://github.com/LedgerHQ/wallet-connect-live-app/assets/5050709/c562afbb-3bb0-4f40-95a7-184bf9c5c128

- We're not preselecting the accounts. The user still has to pick which one he wants. 
- A mock file (src/data/mocks/sessionProposals.ts) contains various proposal examples. Names are descriptive : 

```
    // const proposal = sessionProposals.noSupport;
    // const proposal = sessionProposals.required;
    const proposal = sessionProposals.requiredMissingOne;
    // const proposal = sessionProposals.requiredMissingMultiples;
    // const proposal = sessionProposals.many;
``` 

- The Session screen is also impacted, the account rows are kept consistant with the new ones in proposal. 

SCR-20240516-pzuj

Here is some of the new components: 

SCR-20240516-pxof

SCR-20240516-pzik

### ❓ Context

- **Linked resource(s)**: JIRA: [LIVE-8533](https://ledgerhq.atlassian.net/jira/software/c/projects/LIVE/boards/623?selectedIssue=LIVE-8533)
FIGMA: https://www.figma.com/design/Vl8ot5sWmdHQucuhxRLQRD/Wallet-connect-V2?node-id=1687-6898&t=cD5Oxr353rvBsDxe-0

#### Mobile main flows 

- 1. Old screen, error if chain required is not supported 
- 2. Proposal, 2 required chains, 1 optional
- 3. Missing mandatory accounts on required chains
- 4. All optionals 

https://github.com/LedgerHQ/wallet-connect-live-app/assets/5050709/13c9d885-adde-4b55-8f4c-053ac6e48e4b

### 🚀 Expectations to reach

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-9879]: https://ledgerhq.atlassian.net/browse/LIVE-9879?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LIVE-8533]: https://ledgerhq.atlassian.net/browse/LIVE-8533?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by RamyEB — CHANGES_REQUESTED** [review]

File: src/components/atoms/AccountBalance.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    magnitude: 18,
+    name: network.displayName,
+  };
+  const currencyBalance = formatCurrencyUnit(unit, account.balance, {
+    showCode: true,
+  });
+  return (
+    <Flex flexDirection="column" mr={5}>
+      <Text variant="body" fontWeight="semiBold" color="neutral.c100">
+        {/* NOTE: is there an easy way to get $ (fiat) value of currency from liveapp */}
```
[the remark this answers is not in the document]
Comment (Wozacosta) [reply src/components/atoms/AccountBalance.tsx]: 🧹

File: src/components/screens/SessionProposal.tsx [hunk 2]
```diff
@@ -90,48 +85,64 @@ export default function SessionProposal({ proposal }: Props) {
...
+
+  const chainsWhereNoAccounts = useMemo(
+    () =>
+      accountsByChain
+        .filter((entry) => entry.isSupported)
+        .filter((entry) => entry.accounts.length === 0),
+    [accountsByChain],
   );
 
+  console.log({ chainsWhereNoAccounts, accountsByChain });
```
[the remark this answers is not in the document]
Comment (Wozacosta) [reply src/components/screens/SessionProposal.tsx]: nope, 🧹

**Review by RamyEB — DISMISSED** [review]

File: src/components/screens/sessionProposal/LogoHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        </Flex>
+      </Container>
+    );
+  } else {
+    return (
+      <LogoContainer>
+        <WalletConnectMedium size={30} />
+      </LogoContainer>
+    );
+  }
```
Comment (Justkant) [inline src/components/screens/sessionProposal/LogoHeader.tsx:81]: nit pick to remove the else an rely on the early return
```suggestion
  }
  return (
    <LogoContainer>
      <WalletConnectMedium size={30} />
    </LogoContainer>
  );
```

**Review by RamyEB — APPROVED** [review]

File: src/hooks/useProposal/useProposal.ts [hunk 4]
```diff
@@ -268,6 +268,22 @@ export function useProposal(proposal: ProposalTypes.Struct) {
...
+  const addNewAccounts = useCallback(
+    async (currencies: string[]) => {
+      try {
+        await client.account.request({
+          currencyIds: currencies,
+        });
+        // TODO Maybe we should also select the requested account
+      } catch (error) {
+        console.error("request account canceled by user");
+      }
```
Comment (Justkant) [inline src/hooks/useProposal/useProposal.ts:280]: This is not the only error possible we should show a toast here unless it's cancelled by the user by check the error

**Pull request merged**